chall07/vde-dios.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+    "net/http"
+    "log"
+    "io/ioutil"
+    "fmt"
+    "regexp"
+    "strconv"
+    "strings"
+)
+
+func outputErr(err error){
+    if err != nil {
+        log.Fatalln(err)
+    }
+}
+
+func toHex(rgb [][]string) string {
+    output := ""
+    for i := 1; i < 4; i++ {
+        hex, err := strconv.Atoi(strings.Join(rgb[i],""))
+        outputErr(err)
+        output += fmt.Sprintf("%02x", hex)
+    }
+    return output
+}
+
+func MakeRequest(uri string) string{
+    resp, err := http.Get(uri)
+    outputErr(err)
+    body, err := ioutil.ReadAll(resp.Body)
+    outputErr(err)
+    defer resp.Body.Close()
+    log.Println(string(body))
+    return string(body)
+}
+
+func main() {
+    response := MakeRequest("https://chall07.42madrid.com/")
+    re := regexp.MustCompile(`\d+`)
+    rgb := re.FindAllStringSubmatch(response, -1)
```

Comment:
It seems a little overkill using regular expressions here, but it works.  
I would find more adapted you use [fmt.Sscanf](https://golang.org/pkg/fmt/#Sscanf). As you know exactly how the input should be:
```suggestion
	var id, red, green, blue int
	if _, err := fmt.Sscanf(response, "id=%d,r=%d,g=%d,b=%d - Send your response here: chall07.42madrid.com/?id=<id>&resp=<hex>", &id, &red, &green, &blue); err != nil {
		log.Fatalln(err)
	}
```
I didn't adapat all the code but it gives you a good idea of what to use :)

Later you could simply use `fmt.Sprintf` to format your hexadecimal output` fmt.Sprintf("%02x%02x%02x", red, green, blue)`

Reply:
I will definitely look it at Sscanf!

About regular expressions, I have heard about it several times... Are regexp more resource consuming when compare with other string parsing methods? I mean in most languages (go, python, js...)